standard/tier/tier_utils.lua

```diff
@@ -0,0 +1,177 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Tier/Utils
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+-- module intended to be moved to `Module:Tier` after the old ones usage has been eliminated
+
+local HiddenSort = require('Module:HiddenSort')
+local Logic = require('Module:Logic')
+local Page = require('Module:Page')
+local String = require('Module:StringUtils')
+
+local TierData = mw.loadData('Module:Tier/Data')
+
+local NON_BREAKING_SPACE = '&nbsp;'
+
+local Tier = {}
```

Comment:
I think the tier type functionality should be in a separate module `Module:TierType`

Reply:
we often have a combination of both and they need to be displayed at the same time in a merged string
so at least the display function needs to be shared between both
also imo duplicating a ton of stuff isn't the way to go

Reply:
are there lots of cases where a single variable can hold either a tier or tier type? if not then they are separate enums and should have separate util modules.

If you're worried about code duplication then it might be worth investing in a general enum module. The module is pretty small regardless. (maybe ~100 lines each for Tier/TierType after simplification)

Reply:
- sc/sc2 tournaments list on main page
- potentially matches tables (on player/team subpages)
- potentially portal tournaments pages
- potentially series pages (not on sc2)
- potentially statics pages
- probably a few cases i did not think about

Reply:
In each of those cases do you have separate fields for tier and tierType? (meaning you can have different combiniations of tier/tierType, like minor weekly event, or premier showmatch) Or do you have one variable that can either be a tier or tiertype? (meaning if your event is premier, then it can't also be a showmatch)

Reply:
the only diff between tier and tierType is accessing a different table
why duplicate 9 functions that would do exactly the same?

Reply:
Having good API design and type annotations is more important than duplication. 

Regarding duplication: I count just Tier.getData, Tier.toName, Tier.toShortName, Tier.toStorage as duplicated. 
Tier.read and TierType.read has differences in unknown/default handling.
Tier.Text and TierType.Text has small differences (in the implementation I suggested, which assumes that tiers and tier types aren't interleaved in ordering).
`Tier.isValid` isn't needed because it's the same as `Tier.read(input) ~= nil`
`Tier.getCategory` and `Tier.getLink` aren't super useful

Reply:
i do not really see better api design through this nor better type annos
- Tier.toIdentifier in its current form can also handle input from wikis that have not yet switched to tier+tiertype
   - your suggestion would just always convert it to undefined
- Tier.displaySingle works identical for both tier and tiertype
   - even in youir suggestion (in the other comment) Tier.Text and TierType.Text were identical except for the hiddensort which would not work anyways if we ever had a wiki with > 10 tiertypes (e.g. sc2 has 13 ...)
- having Tier.isValid as a function is imo preferably compared to doing a Tier.raw(Tier.toIdentifier(...), ...) ~= nil
   - toIdentifier is not intended to return nil for unknown values it is just to normalize the input in a way that we do not need a ton of aliases

Reply:
it still breaks/changes lots of functionality the module is intended to have